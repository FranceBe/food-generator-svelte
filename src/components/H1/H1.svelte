<script>
    export let title
</script>
<style>
    div {
        display: flex;
        align-items: center;
    }
    h1 {
        font-size: 1.875rem;
        text-align: center;
        text-shadow: 1px 1px #EBA789;
    }
 </style>

<div><h1>{title}</h1><slot></slot></div>
