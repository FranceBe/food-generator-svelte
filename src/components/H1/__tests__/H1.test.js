import { cleanup, render } from '@testing-library/svelte'
import H1 from 'components/H1/H1.svelte'

describe('H1', () => {
  afterEach(() => {
    cleanup()
  })
  it('should match snapshot', () => {
    const { container } = render(H1, {
      props: {
        title: 'My title',
      }
    })
    expect(container).toMatchSnapshot()
  })
  it('should render an h1 tag', () => {
    const { container, getAllByText } = render(H1, {
      props: {
        title: 'My title',
      }
    })
    expect(container.getElementsByTagName('h1')).toHaveLength(1)
    expect(getAllByText('My title').length).toBe(1)
  })
})
