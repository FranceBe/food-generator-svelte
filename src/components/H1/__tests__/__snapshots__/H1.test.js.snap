// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 should match snapshot 1`] = `
<body>
  <div>
    <h1>
      My title
    </h1>
  </div>
</body>
`;
